frontend/src/views/SecretOverviewPage/SecretOverviewPage.tsx

```diff
@@ -390,41 +394,44 @@ export const SecretOverviewPage = () => {
           <div className="flex items-center justify-between">
             <FolderBreadCrumbs secretPath={secretPath} onResetSearch={handleResetSearch} />
             <div className="flex flex-row items-center justify-center space-x-2">
-              <DropdownMenu>
-                <DropdownMenuTrigger asChild>
-                  <IconButton
-                    ariaLabel="Environments"
-                    variant="plain"
-                    size="sm"
-                    className="mr-2 flex w-11 items-center justify-center overflow-hidden border border-mineshaft-600 bg-mineshaft-800 p-0 hover:border-primary/60 hover:bg-primary/10"
-                  >
-                    <Tooltip content="Choose visible environments" className="mb-2">
-                      <FontAwesomeIcon icon={faList} />
-                    </Tooltip>
-                  </IconButton>
-                </DropdownMenuTrigger>
-                <DropdownMenuContent align="end">
-                  <DropdownMenuLabel>Choose visible environments</DropdownMenuLabel>
-                  {userAvailableEnvs.map((avaiableEnv) => {
-                    const { id: envId, name } = avaiableEnv;
+              {userAvailableEnvs.length > 0 && (
+                <DropdownMenu>
+                  <DropdownMenuTrigger asChild>
+                    <IconButton
+                      ariaLabel="Environments"
+                      variant="plain"
+                      size="sm"
+                      className="mr-2 flex h-10 w-11 items-center justify-center overflow-hidden border border-mineshaft-600 bg-mineshaft-800 p-0 hover:border-primary/60 hover:bg-primary/10"
```

Comment:
Spacing looks a bit out of place here. Can we remove the `mr-2` class to make it more uniformed?

![CleanShot 2024-03-12 at 16 07 10@2x](https://github.com/Infisical/infisical/assets/62331820/fd7dd3f1-3d3f-4759-88c1-7bcf991cd46a)